src/GNDStk/Node/test/add.test.cpp

```diff
@@ -0,0 +1,124 @@
+
+// fixme
+// This file tests all of Node's add() functions, but could use
+// some reorganization in order to make it all much clearer.
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+using namespace njoy::GNDStk;
+
+
+// ------------------------
+// dimensions2d; example
+// for metadata
+// ------------------------
+
+struct dimensions2d {
+   int rows;
+   int cols;
+   dimensions2d() : rows(10), cols(10) { }
+   dimensions2d(const int r, const int c) : rows(r), cols(c) { }
+};
+
+std::ostream &operator<<(std::ostream &s, const dimensions2d &dim)
+{
+   return s << dim.rows << ',' << dim.cols;
+}
+
+
+// ------------------------
+// dim2d; example for
+// nodes
+// ------------------------
+
+struct dim2d {
+   int rows;
+   int cols;
+   ///   dim2d() : rows(10), cols(10) { }
+   dim2d(const int r, const int c) : rows(r), cols(c) { }
+};
+
+void convert(const dim2d &d, Node<> &out)
+{
+   out.name = "unused";
+   out.add("numberOfRows",d.rows);
+   out.add("numberOfColumns",d.cols);
+   out.add("firstExampleChild");
+   out.add("secondExampleChild");
+}
+
+
+// ------------------------
+// Scenario
+// ------------------------
+
+SCENARIO("Testing GNDStk Node add()") {
+
+   Node<> n;
+   n.name = "name";
+   // debug = true;
+
+   // metadata, string key, various value types
+   n.add(std::make_pair("a","b"));
```

Comment:
Yes, and the short version of the response is that I've added enough, since writing the bulk of the tests, that the tests themselves probably need a substantial revamp. Note that some things end up essentially being tested as part of another test; for example, add() ends up being tested in Node's copy.test.cpp as well.

Reply:
For now, I did go ahead and add just a bit to the file in question.